src/nni_manager/training_service/reusable/trialDispatcher.ts

```diff
@@ -578,6 +578,15 @@ class TrialDispatcher implements TrainingService {
 
         }
     }
+    
+    private async initEnvironments(): Promise<void> {
+        const environmentService = component.get<EnvironmentService>(EnvironmentService);
+        const number = environmentService.getInitializeEnvironmentNumber();
+        this.log.info(`Initialize environments total number: ${number}`);
+        for (let index = 0; index < number; index++) {
+            await this.requestEnvironment();
```

Comment:
It may need to be async call, and wait together. So that it can be faster.

Reply:
I have a look on code, remote machine starts all envs in config, so it has no effect. But others may implement with a sync way, so it should be async, and wait all here.

Reply:
didn't get the point, do you mean `await Promise.all()`?

Reply:
agree to do it concurrently (Promise.all) if possible.

Reply:
Remote machine is scheduled sequentially, if request environment simultaneously, there might be potential issue?

Reply:
if it can be initialized sequentially, it looks no benefit to initialize before running. If there are 1000 machines, it may need 10 minutes, and first runner timeout already.